src/cpu-template-helper/src/fingerprint/compare.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::ValueEnum;
+
+use crate::fingerprint::{Fingerprint, FingerprintField};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// Difference detected between source and target.
+    #[error("Difference detected between source and target: {0}")]
+    DiffDetected(String),
+}
+
+pub fn compare(
+    source: Fingerprint,
+    target: Fingerprint,
+    filters: Vec<FingerprintField>,
+) -> Result<(), Error> {
+    let source = serde_json::to_value(source).unwrap();
+    let target = serde_json::to_value(target).unwrap();
+
+    let mut diff_str = String::new();
+    for filter in filters {
+        let field = filter.to_possible_value().unwrap();
+        let name = field.get_name();
+
+        if source[name] != target[name] {
+            diff_str.push_str(&format!(
+                "\n{{\n  \"{}\": {{\n    \"source\": {},\n    \"target\": {}\n  }}\n}}",
```

Comment:
If the `guest_cpu_config` is the one that is different, we print both full configs. Is it going to be easy for the users to see what the actual difference is (if it's just one bit)? Will they need to use an additional tool to understand that?

Reply:
The main difficulty is that the structure of `guest_cpu_config` is vectors of modifiers (especially nested for CPUID), not key-value pairs.

Reply:
That's reasonable. Perhaps we could give some suggestions in the docs on how to narrow down the difference?

Reply:
But we can reuse the strip feature even now.

Reply:
Sample result (flipping 1 bit of CPUID leaf 0 / EAX):
```
$ cpu-template-helper fingerprint compare --prev fingerprint.json --curr fingerprint_new.json --filters guest_cpu_config
Error: Difference detected between source and target:
{
  "name": "guest_cpu_config",
  "prev": {
    "cpuid_modifiers": [
      {
        "flags": 0,
        "leaf": "0x0",
        "modifiers": [
          {
            "bitmap": "0b00000000000000000000000000011011",
            "register": "eax"
          }
        ],
        "subleaf": "0x0"
      }
    ],
    "msr_modifiers": []
  },
  "curr": {
    "cpuid_modifiers": [
      {
        "flags": 0,
        "leaf": "0x0",
        "modifiers": [
          {
            "bitmap": "0b00000000000000000000000001011011",
            "register": "eax"
          }
        ],
        "subleaf": "0x0"
      }
    ],
    "msr_modifiers": []
  }
}
```